### PR TITLE
Add Cache for Message Serverless Function 

### PR DESCRIPTION
**Pull request by pzalep1**
This PR adds logic to cache the result of the serverless function that is used to display the message banner in the client. 

This PR also added sentry for error logging and the env file.

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -1,7 +1,32 @@
...
+import { reportError } from './shared/SentryConnector';
+const APP_STATUS = process.env.APP_STATUS_URI;
 // ----------------------------------------------------------------------------------
 // Initializations
 // ----------------------------------------------------------------------------------
 
 // ----------------------------------------------------------------------------------
 ExpressDriver.start();
+
+async function fillCache() {
```
Comment (nvisal1) [inline src/app.ts]: Can you move this logic into the ServerlessCache module? It will be easier to separate when we move away from using the Gateway.

Comment (pzalep1) [reply src/app.ts]: Resolved with 14af224

File: src/cache.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+     }
+     static get cachedValue() {
+         return this._cacheValue;
+     }
+
+    static fillCache() {
+        try {
+            request(APP_STATUS, function(error, response, body) {
+                ServerlessCache.cachedValue = body;
+                console.log(body);
```
Comment (nvisal1) [inline src/cache.ts]: remove dev log

File: src/cache.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+     static set cachedValue(val) {
+        this._cacheValue = val;
+     }
+     static get cachedValue() {
+         return this._cacheValue;
+     }
+
+    static fillCache() {
+        try {
+            request(APP_STATUS, function(error, response, body) {
```
Comment (DonSeannelly) [inline src/cache.ts]: I think that the request API might be stream based, so the try catch won’t actually wrap the error. We should try attaching an event listener for errors.

File: src/cache.ts (hunk 2 above)
Comment (pzalep1) [reply src/cache.ts]: Resolved with 72282f3

File: src/cache.ts (hunk 3 above)
Comment (pzalep1) [reply src/cache.ts]: Resolved with 80dde11

**Comment by pzalep1** [comment]
The CodeFactor Error is coming from the SentryConnector that we copied from Learning Object Service. What should I do to stop the tests from failing?

**Comment by DonSeannelly** [comment]
> The CodeFactor Error is coming from the SentryConnector that we copied from Learning Object Service. What should I do to stop the tests from failing?

I think you should be able to remove the assignment from the declaration, and just leave the type annotation there. That might fix it.

**Comment by pzalep1** [comment]
> > The CodeFactor Error is coming from the SentryConnector that we copied from Learning Object Service. What should I do to stop the tests from failing?
> 
> I think you should be able to remove the assignment from the declaration, and just leave the type annotation there. That might fix it.

That did it. Thanks!

File: src/app.ts [hunk 4]
```diff
@@ -1,7 +1,22 @@
...
+async function setCacheInterval() {
+    setInterval(() => {
+        ServerlessCache.fillCache();
+    // tslint:disable-next-line: align
+    }, 300000); // 5 minute interval
+}
+if (environment === 'development') {
+    ServerlessCache.fillCache();
+    setCacheInterval();
+}
```
Comment (nvisal1) [inline src/app.ts]: This condition needs to be switched to run in production

**Pull request merged**